example/oven_fsm.dart

```diff
@@ -0,0 +1,228 @@
+/// Copyright (C) 2023 Intel Corporation
+/// SPDX-License-Identifier: BSD-3-Clause
+///
+/// oven_fsm.dart
+/// A simple oven FSM implementation using ROHD.
+///
+/// 2023 February 13
+/// Author: Yao Jing Quek <yao.jing.quek@intel.com>
+///
+
+// ignore_for_file: avoid_print
+
+// Import the ROHD package.
+import 'package:rohd/rohd.dart';
+
+// Import the counter module implement in example.dart.
+import './example.dart';
+
+// Enumerated type named `OvenState` with four possible states:
+// `standby`, `cooking`,`paused`, and `completed`.
+enum OvenState { standby, cooking, paused, completed }
+
+// One-hot encoded `Button` that extends from Const() value.
+// Represent start, pause, and resume as constant value `00`, `01`,
+// and `10` respectively.
+class Button extends Const {
```

Comment:
I think we can replace both `Button` and `LEDLight` with `enum`s with values:
https://dart.dev/guides/language/language-tour#enumerated-types

I think it would look cleaner than having these `Const` subclasses